### DISCUSSION TITLE
Time / Date / Hardware

### DISCUSSION THREAD

**Discussion by MyLive** [ipxe/ipxe#1292] [Ideas]
Hello @ all,

Linux and Windows needs a correct time / date for Installation, to verify the Certificates.
i have the wish for integrating some Commands :-)
- set_HW_Clock
- date ( English / German Format :-) ) which shows the current Time / Date of the System

Nice but not realy needed :-P is a way to use the Display of Date in a Menu or predefined Positon ( like from top - from left ) with the Time + Date which will refreshed every second :-)

Best regards,
René :-)

**Comment by mcb30** [comment]
Use https://ipxe.org/cmd/ntp

**Reply by MyLive** [reply]
But this does not set the Hardware Clock ....

**Reply by mcb30** [reply]
Correct. iPXE is not a general-purpose operating system, and you should not expect to find every possible tool available in the iPXE environment. 

iPXE needs an approximately correct clock for only one purpose: validating the timestamps on digital signatures. Reading the hardware clock is usually good enough for this. When the hardware clock is wrong, the `ntp` command lets us work around the problem. 

We don't ever need the ability to set the hardware clock. Use the tools within your booted operating system to do this.

**Comment by MyLive** [comment]
But If the time is wrong, e.g. Linux can Not BE installed due to unvaild certificates :-/ .. this is what i Like to fix with it. . 

⁣Gesendet mit BlueMail ​

**Reply by mcb30** [reply]
> But If the time is wrong, e.g. Linux can Not BE installed due to unvaild certificates

Yes it can. Use `ntp` to correct the time in iPXE so that you can pass the relevant certificate validity checks within iPXE while loading the installer. Then run the installer.

You can use NTP to set the time within the installer or, if you don't have an NTP client in your installer's initrd image, you can pass iPXE's `${unixtime}` setting in the kernel command line as a way to set an initial system time.

**Reply by NiKiZe** [reply]
Don't your installer have a ntp client? If so report that to your distro of choice.

On another note, when replying to discussions don't use mail, instead visit the thread, and answer below the correct post.

**Reply by marvinhinz** [reply]
Same issue here, we use ipxe to install pos devices. IPXE passes a custom url to the installer, that serves an individual preseed configuration file via https. The target device has a wrong hardware clock out of the box, and it seems there is no way to set the clock in the installer or pass it via kernel args (afaik). @mcb30 do you know an arg that accepts the ${unixtime}?

**Reply by NiKiZe** [reply]
The kernel probably does not, but scripts or the preseed configuration might, but that would be distribution specific.
If however you could inject a script in the initrd, than that could be used to set the time.
